solr/solrj/src/java/org/apache/solr/common/cloud/SolrZkClient.java

```diff
@@ -323,42 +370,65 @@ public Boolean exists(final String path, boolean retryOnConnLoss)
    */
   public List<String> getChildren(final String path, final Watcher watcher, boolean retryOnConnLoss)
       throws KeeperException, InterruptedException {
+    List<String> result = null;
     if (retryOnConnLoss) {
-      return zkCmdExecutor.retryOperation(() -> keeper.getChildren(path, wrapWatcher(watcher)));
+      result = zkCmdExecutor.retryOperation(() -> keeper.getChildren(path, wrapWatcher(watcher)));
     } else {
-      return keeper.getChildren(path, wrapWatcher(watcher));
+      result = keeper.getChildren(path, wrapWatcher(watcher));
+    }
+
+    metrics.childFetches.increment();
```

Comment:
before executing the op, similar to other places?

Reply:
yeah; after might actually be preferable, but whether before/after it should be consistent/intentional. This is what I was thinking of above:

>based on where the stats are actually incremented, we're currently not capturing any info about connection errors/retries

If "before" you could have tons of stats incremented even with every attempt failing/throwing an exception ...